plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/TestSqlServerTypeMapping.java

```diff
@@ -66,6 +95,124 @@ public void testVarbinary()
                 .execute(getQueryRunner(), trinoCreateAsSelect("test_varbinary"));
     }
 
+    /**
+     * @see #testTimestampCoercion
```

Comment:
If you use `SqlDataTypeTest` for the new test code, you can merge the two test methods into one

@hashhar can help if needed

Reply:
That was super easy with `DataTypeTestToSqlDataTypeTestConverter`, thanks!

Reply:
The second method doesn't use a data provider. Does it make sense to test literal timestamps with different session time zones?

Reply:
I do not see why it would make sense for one method and not the other. Or actually why we have two methods at all. @findepi ?

Reply:
We have two methods because once upon a time (not very long ago), we did not have `SqlDataTypeTest` class.
So some test cases where covered by `DataTypeTest`. Since it couldn't handle timestamps with precision >9, there were manually crafted tests.... the `SqlDataTypeTest` archetype.

Now it no longer makes sense to keep this as two methods.

> Does it make sense to test literal timestamps with different session time zones?

Well... one could argue that time zone doesn't matter (and that's true today!)
But 1) it did matter in the past and 2) the implementation could do something silly, so testing with some session zones covers for that too.
(This is also why we're using these insanely weird time zones for JVM zone in tests).

Reply:
I moved most tests into a single test function. I left those that test `DATETIME2` explicitly in a separate test since they don't use the session with custom timezones.